### PR TITLE
refactor(framework): replace fastjson with jackson

### PR DESCRIPTION
**Pull request by halibobo1205**
## **User description**
Remove the Fastjson dependency entirely and replace it with Jackson-backed drop-in wrappers (JSON, JSONObject, JSONArray, JSONException) that preserve the same public API surface.

Motivation:
- Fastjson has a history of critical CVEs and is no longer actively maintained for 1.x
- Jackson-databind 2.18.6 addresses CVE GHSA-72hv-8253-57qq

Core changes (common module):
- Add org.tron.json.{JSON, JSONObject, JSONArray, JSONException} wrappers backed by a shared Jackson ObjectMapper configured to match Fastjson 1.x parsing/serialization defaults:
  - Unquoted field names and single-quoted strings (lenient parsing)
  - BigDecimal for floats, case-insensitive property matching
  - Null fields omitted (matches Fastjson default)
- Type-safe accessors: getBoolean/getLong/getDouble/getIntValue/ getLongValue/getBigDecimal throw JSONException on invalid text instead of silently returning 0/false
- parseObject(String) guards against ClassCastException on non-object JSON roots; parseArray handles whitespace-only input
- parseObject(String, Class) delegates to parseObject/parseArray for wrapper types to avoid silent field loss via ObjectMapper
- Upgrade jackson-databind 2.18.3 → 2.18.6

HTTP servlet changes (framework module):
- Swap import from com.alibaba.fastjson → org.tron.json across all HTTP API servlets, JSON-RPC layer, and event/log parsers
- No changes to request/response JSON structure — existing API contracts are preserved

Test changes:
- Add BaseHttpTest base class managing Args lifecycle, Wallet mock, MINIMAL_TX constant, and request/response factory methods (postRequest, getRequest, newResponse)
- 44 servlet test classes refactored to extend BaseHttpTest, eliminating ~1400 lines of duplicated boilerplate
- Strengthen weak assertNotNull checks to content-based assertions: assertTrue(contains("raw_data")) for transaction servlets, assertTrue(contains("blockID")) for block queries, etc.
- Add Mockito verify for wallet service calls in query servlets to catch request-to-service mapping regressions
- Fix test environment: initialize Args from config-test.conf (maxMessageSize) and use MINIMAL_TX with raw_data to prevent NPE in Util.printCreateTransaction
- Add JsonCompatibilityFuzzTest: 500-round fuzz covering round-trip serialization, BigDecimal/BigInteger precision, deep nesting, unicode, and boundary values
- Use SecureRandom for fuzz test randomization

Build:
- Remove fastjson from common/build.gradle dependencies
- Update gradle/verification-metadata.xml for jackson 2.18.6

**Follow up**

**Extra details**

___

## **CodeAnt-AI Description**
**Switch JSON handling to the new Jackson-backed wrappers while keeping the same API shape**

### What Changed
- JSON input now accepts the same flexible formats as before, including unquoted field names, single quotes, and trailing commas
- JSON output keeps omitting null fields, and pretty-printed output remains available
- Invalid numeric values now fail clearly instead of being treated as 0 or false
- HTTP requests, JSON-RPC, and event parsing now use the new JSON wrappers without changing response formats
- The VM trace string output now uses the default formatted form

### Impact
`✅ Safer JSON parsing`
`✅ Clearer invalid input errors`
`✅ Same API responses after JSON library swap`
<details>
<summary><strong>💡 Usage Guide</strong></summary>

### Checking Your Pull Request
Every time you make a pull request, our system automatically looks through it. We check for security issues, mistakes in how you're setting up your infrastructure, and common code problems. We do this to make sure your changes are solid and won't cause any trouble later.

### Talking to CodeAnt AI
Got a question or need a hand with something in your pull request? You can easily get in touch with CodeAnt AI right here. Just type the following in a comment on your pull request, and replace "Your question here" with whatever you want to ask:
<pre>
<code>@codeant-ai ask: Your question here</code>
</pre>
This lets you have a chat with CodeAnt AI about your pull request, making it easier to understand and improve your code.

#### Example
<pre>
<code>@codeant-ai ask: Can you suggest a safer alternative to storing this secret?</code>
</pre>

### Preserve Org Learnings with CodeAnt
You can record team preferences so CodeAnt AI applies them in future reviews. Reply directly to the specific CodeAnt AI suggestion (in the same thread) and replace "Your feedback here" with your input:
<pre>
<code>@codeant-ai: Your feedback here</code>
</pre>
This helps CodeAnt AI learn and adapt to your team's coding style and standards.

#### Example
<pre>
<code>@codeant-ai: Do not flag unused imports.</code>
</pre>

### Retrigger review
Ask CodeAnt AI to review the PR again, by typing:
<pre>
<code>@codeant-ai: review</code>
</pre>

### Check Your Repository Health
To analyze the health of your code repository, visit our dashboard at [https://app.codeant.ai](https://app.codeant.ai). This tool helps you identify potential issues and areas for improvement in your codebase, ensuring your repository maintains high standards of code health.

</details>

### CODE REVIEW

**Comment by halibobo1205** [comment]
@codeant-ai: review

**Comment by halibobo1205** [comment]
@codex review

File: common/src/main/java/org/tron/common/utils/JsonUtil.java [hunk 1]
```diff
@@ -1,14 +1,16 @@
 package org.tron.common.utils;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.json.JsonMapper;
 import org.springframework.util.StringUtils;
 
 public class JsonUtil {
 
+  private static final ObjectMapper om = new JsonMapper();
```
[the remark this answers is not in the document]
Comment (halibobo1205) [reply common/src/main/java/org/tron/common/utils/JsonUtil.java:9]: @codeant-ai: Compare this change with the base branch,  and re-review it

**Comment by halibobo1205** [comment]
@codeant-ai : review

**Comment by halibobo1205** [comment]
@codex review

**Comment by halibobo1205** [comment]
@codeant-ai: review

**Comment by halibobo1205** [comment]
@CodeAnt-AI: review
